xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/Code/NotificationsEmailPreferencesMacro.xml

```diff
@@ -440,33 +440,32 @@
-            #set ($objPref = $targetDocument.getObject('XWiki.Notifications.Code.NotificationEmailPreferenceClass'))
-            #if ("$!objPref" != '')
-              #set ($objInterval = $objPref.getValue('interval'))
-              #if ("$!objInterval" != '')
-                #set ($interval = $objInterval)
+            #if ($wikimacro.parameters.target == 'user')
+              #set ($interval = $services.notification.preferences.getInterval($targetUser))
+            #else
+              #set ($interval = 'DAILY')
+              #set ($targetDocument = $xwiki.getDocument($services.model.createDocumentReference('', ['XWiki', 'Notifications', 'Code'], 'NotificationAdministration')))
+              #set ($objPref = $targetDocument.getObject('XWiki.Notifications.Code.NotificationEmailPreferenceClass'))
+              #if ("$!objPref" != '')
+                #set ($objInterval = $objPref.getValue('interval'))
+                #if ("$!objInterval" != '')
+                  #set ($interval = $objInterval)
+                #end
               #end
             #end
             &lt;select class="notificationEmailInterval"&gt;
-              &lt;option value="hourly" #if($interval == 'hourly')selected="selected"#end&gt;$escapetool.xml($services.localization.render('XWiki.Notifications.Code.NotificationEmailPreferenceClass_interval_hourly'))&lt;/option&gt;
-              &lt;option value="daily"  #if($interval == 'daily')selected="selected"#end&gt;$escapetool.xml($services.localization.render('XWiki.Notifications.Code.NotificationEmailPreferenceClass_interval_daily'))&lt;/option&gt;
-              &lt;option value="weekly" #if($interval == 'weekly')selected="selected"#end&gt;$escapetool.xml($services.localization.render('XWiki.Notifications.Code.NotificationEmailPreferenceClass_interval_weekly'))&lt;/option&gt;
-              &lt;option value="live" #if($interval == 'live')selected="selected"#end&gt;$escapetool.xml($services.localization.render('XWiki.Notifications.Code.NotificationEmailPreferenceClass_interval_live'))&lt;/option&gt;
+              &lt;option value="HOURLY" #if($interval == 'HOURLY')selected="selected"#end&gt;$escapetool.xml($services.localization.render('XWiki.Notifications.Code.NotificationEmailPreferenceClass_interval_hourly'))&lt;/option&gt;
+              &lt;option value="DAILY"  #if($interval == 'DAILY')selected="selected"#end&gt;$escapetool.xml($services.localization.render('XWiki.Notifications.Code.NotificationEmailPreferenceClass_interval_daily'))&lt;/option&gt;
+              &lt;option value="WEEKLY" #if($interval == 'WEEKLY')selected="selected"#end&gt;$escapetool.xml($services.localization.render('XWiki.Notifications.Code.NotificationEmailPreferenceClass_interval_weekly'))&lt;/option&gt;
+              &lt;option value="LIVE" #if($interval == 'LIVE')selected="selected"#end&gt;$escapetool.xml($services.localization.render('XWiki.Notifications.Code.NotificationEmailPreferenceClass_interval_live'))&lt;/option&gt;
```

Comment:
Sounds like it could have been refactored with a foreach loop

Reply:
Hmmm, actually does it mean there's a bug right now and the interval is not properly selected in the UI? Sounds a bit surprising

Reply:
There was no bug: the interval was stored in lower case in the user profile, and was then converted to upper case by the IntervalUsersManager (reference: [IntervalUsersManager.java#L130](https://github.com/xwiki/xwiki-platform/blob/101becab4069ae9ccdc8c2f2c4edc562645e152b/xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-notifiers/xwiki-platform-notifications-notifiers-api/src/main/java/org/xwiki/notifications/notifiers/internal/email/IntervalUsersManager.java#L130)) but was always retrieved as is from the UI (no case conversion). Now the API `NotificationEmailUserPreferenceManager#getInterval` returns only upper case values, the goal here is to align the case with what the API returns (same approach as what we have with the diffType option values).

Reply:
ok I understand the idea, but not entirely comfortable with it: it means that an upgraded wiki might have both lower case and upper case stored interval values, and it might possibly break some external code that are reading those values directly from the xobject. 

I agree that the fact the case of the stored data and what's returned by the API should be same, but I'm afraid such change would be possibly breaking and would need some kind of migrations. To be honest I think I'd have been ok with that change if it wouldn't be for 14.10. 

Would need opinion of other committers here, @manuelleduc  @vmassol wdyt?

Reply:
I brought more changes:
- Converted interval value to upper case for existing instances where it would be stored in lower case in an xobject.
- Added foreach loop.
- Added a test case (but cannot run the test suite successfully on my side for now (see error below, any help welcome).
- Removed unncessary test on null value already covered by StringUtils.isNotBlank.

```
org.xwiki.component.manager.ComponentLookupException: Can't find descriptor for the component with type [org.xwiki.model.reference.DocumentReferenceResolver<java.lang.String>] and hint [null]
	at org.xwiki.component.embed.EmbeddableComponentManager.getInstance(EmbeddableComponentManager.java:212)
	at org.xwiki.component.embed.EmbeddableComponentManager.getInstance(EmbeddableComponentManager.java:189)
	at org.xwiki.notifications.preferences.internal.email.DefaultNotificationEmailUserPreferenceManagerTest.setUp(DefaultNotificationEmailUserPreferenceManagerTest.java:62)
```